### DISCUSSION TITLE
Version command not running

### DISCUSSION THREAD

**Discussion by vani2620** [alire-project/alire#1303] [Support]
I tried building the project from source and while it seemed to have built successfully, running `alr version` doesn't seem to work. I'm using 64-bit Windows. The error I get reads

Program 'alr.exe' failed to run: The specified executable is not a valid application for this OS platform.At line:1
char:1
+ alr version
+ ~~~~~~~~~~~.
At line:1 char:1
+ alr version
+ ~~~~~~~~~~~
    + CategoryInfo          : ResourceUnavailable: (:) [], ApplicationFailedException
    + FullyQualifiedErrorId : NativeCommandFailed

**Comment by mosteo** [comment]
How did you do the build?

For this kind of question I think is more appropriate to use the Discussions, at least unless some clear bug is identified. I'm going to convert this issue but I don't think it changes anything wrt to subscriptions and etc.

**Reply by vani2620** [reply]
I built it using the latest version of the GNAT Community Edition compiler (20210519-103), and it seemed to have worked perfectly, except running alr version in the terminal gave me the above error. Unfortunately not sure of what exactly went wrong.

**Reply by vani2620** [reply]
I did try installing it prior to the build, too, and I ran into the same issue, although that at least worked when I ran the .exe file. 

P.S. Apologies for the late replies. I...forgot to check my email.

**Reply by mosteo** [reply]
I still need more details on how you did the build, more precisely did you do one of:
- `alr build` 
- `gprbuild -P alr_env.gpr`
?

Also, what's the output of `gnat --version`?

**Reply by vani2620** [reply]
I used the `gprbuild -P alr_env.gpr` command for the build.

`gnat --version` outputs the following:

> GNAT Community 2021 (20210519-103)
Copyright (C) 1996-2021, Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

**Reply by mosteo** [reply]
Can't say how could this happen. Truncated executable (out of space?), another `alr` for other platform downloaded elsewhere but also in path?

**Reply by vani2620** [reply]
...It ended up being the latter. I installed it via WSL2 on Ubuntu-22.04 and it works perfectly fine.
